### ISSUE TITLE
Identify and remove dead code

### ISSUE THREAD

**Issue by nijel** [translate/translate#4018]
There is probably quite some dead code lying around. It would be great if long time contributors @unho or @julen could point to such so that it can be properly investigated and possibly removed to avoid need to maintain code nobody uses these days.

Also do you know any other major users of the toolkit besides amaGama, Virtaal and Pootle? Those are probably broken by move to Python 3, but we should not break them more than necessary.

Inspired by https://github.com/translate/translate/pull/3967#pullrequestreview-391292203

**Comment by unho** [comment]
One of the big problems with Translate Toolkit is that it keeps together unrelated code. For example we have:
* the storage (used by third party projects and Translate Toolkit),
* the converters,
* other tools and scripts,
* the language filters (for quality check),
* the language data (including filter customizations),
* code used by some other Translate projects, like search and indexing,
* ...
There was a proposal long ago about splitting these into separate projects, but we never got there. I mean, it is not as much as dropping unused code (which probably is being used anywhere), but about moving out to separate apps parts of the codebase that are usually not used.

**Comment by nijel** [comment]
For some things, it's certainly useful to keep them together (storage and converters) as their code is usually heavily coupled. 

My knowledge of rest is minimal, but I'm pretty sure there are many things not used anywhere. This happens to every bigger project as the code tends to accumulate...

**Comment by julen** [comment]
I'm almost certainly sure the code for L20N support can be considered dead, because [L20N is deprecated](https://github.com/l20n) on its own (later on superseded by [Fluent](https://projectfluent.org/)). Likewise, given that Mozilla is actively moving to Fluent, and the Moz-specific tools were maintained as part of the mozilla.locamotion.org effort (now dead), Mozilla-related converters could be considered dead too.

**Comment by unho** [comment]
I agree with dropping the parts @julen talks about in his last comment because L20N was never actually of widespread usage and those tools are now useless. And if anyone actually needs those then an older version of TTK can be used. But I still would keep storage and converters for DTD and other formats that were widely used by Mozilla to ease converting translations (I sometimes convert translations to TMX so they can be reused among translation projects, or used in any other way like extracting terminology).

**Comment by nschonni** [comment]
Might be good to create a CI job (or jobs) that would pull down and run the dependent projects to look for regressions. Don't think it would need to block merging things or tested across all Python versions

**Comment by unho** [comment]
Another thing I am not sure why it is in TTK is `tmserver`. It doesn't seem to be used in any other Translate project and it kinda overlaps with amaGama. Maybe @julen can provide some info on this.

**Comment by julen** [comment]
@unho that's used in Virtaal for its local TM feature.

**Comment by nijel** [comment]
> Might be good to create a CI job (or jobs) that would pull down and run the dependent projects to look for regressions. Don't think it would need to block merging things or tested across all Python versions

I don't think it should be done here. It would mostly duplicate the CI logic of these projects and would be hard to maintain - it's better to maintain this at the individual projects.

Also, Virtaal and Pootle AFAIK do not support Python 3, so there is no point to test them against current translate-toolkit until this is addressed. For Weblate we do have testing against translate-toolkit master branch in our CI.

The cleanup is always matter of breakage we're willing to introduce. I intended not to remove any code which we know is used (still this might cause some breakages for uses we do not know).

**Comment by nijel** [comment]
I've done some cleanup PRs.

I didn't touch tmserver, because I don't have strong opinion here - the program was provided by translate-toolkit for years, so there might be others relying on that. Merging it into Virtaal doesn't sound right for such use case, so either it should stay as it is or be a standalone package. It overlaps with amaGama, but it doesn't require a database server, so I believe it can have its use case.

**Comment by unho** [comment]
If it were up to me I would spun `tmserver` out to a separate project.

**Comment by nijel** [comment]
That seems like a best approach. The only downside I see is that it will more or less become abandoned as the only user is Virtaal, which doesn't see much activity these days. When the code is inside translate-toolkit, it will be kept updated to work, but as a separate module it will be forgotten. I'm willing to help with the split, but I have no interest in maintaining the code in a standalone package.

**Comment by unho** [comment]
Then lets keep it in TTK for now.

**Comment by nijel** [comment]
Okay, removal pull request is here: https://github.com/translate/translate/pull/4069

**Comment by nijel** [comment]
https://github.com/translate/translate/pull/5206 is pull request to remove Mozilla build tools.

**Comment by nijel** [comment]
I've decided to remove `tmserver` and related code in https://github.com/translate/translate/pull/6048. If there is some interest in it, let's resurrect it in a separate package, but it definitely needs some maintenance, and nobody was interested in past years.
